File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/TotalMatchedFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class TotalMatchedFunction implements ProgressFunction {

    @Override
    public OptionalDouble function(SettledBet bet) {
        Double matchedAmount = bet.getBetAction().getMarket().getMatchedAmount();
        return matchedAmount == null ? OptionalDouble.empty() : OptionalDouble.of(matchedAmount);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/TotalMatchedFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component

@Component
object TotalMatchedFunction : ProgressFunction {

    override fun invoke(bet: RealizedBet): Double {
        return bet.market.matchedAmount
    }

}
